lib/staging/tls/tls.cpp

```diff
@@ -1058,6 +1101,12 @@ void Server::wait_for_connection(const ConnectionHandler& handler) {
                 // new connection, pass to handler
                 auto* ip = BIO_ADDR_hostname_string(peer.get(), 1);
                 auto* service = BIO_ADDR_service_string(peer.get(), 1);
+
+                if (m_tls_key_interface) {
+                    const auto port = std::stoul(service);
+                    keylog_server = std::make_unique<TlsKeyLoggingServer>(std::string(m_tls_key_interface), port);
```

Comment:
SSL_set_ex_data could be used here to set the server for this connection.
logging path could also be set
Better to set it in the connection object with the actual SSL object.
(line 740 in ServerConnection constructor)

Reply:
Done